enderio-base/src/main/java/com/enderio/base/client/renderers/GraveRenderer.java

```diff
@@ -23,30 +25,34 @@
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.level.block.entity.BlockEntity;
 
+import javax.annotation.Nonnull;
+
 //renders grave as a playerskull
 public class GraveRenderer implements BlockEntityRenderer<BlockEntity> {
-    private BlockEntityRendererProvider.Context context;
+    private final BlockEntityRendererProvider.Context context;
 
     public GraveRenderer(BlockEntityRendererProvider.Context context) {
         this.context = context;
     }
 
     @Override
-    public void render(BlockEntity pBlockEntity, float pPartialTicks, PoseStack pMatrixStack, MultiBufferSource pBuffer, int pCombinedLight,
+    public void render(@Nonnull BlockEntity pBlockEntity, float pPartialTicks, PoseStack pMatrixStack, @Nonnull MultiBufferSource pBuffer, int pCombinedLight,
         int pCombinedOverlay) {
         GraveBlockEntity grave = (GraveBlockEntity) pBlockEntity;
         Direction direction = null;//TODO if we make the grave rotatable
         SkullModelBase skullmodelbase = new SkullModel(this.context.bakeLayer(ModelLayers.PLAYER_HEAD));
-        RenderType[] rendertype = new RenderType[] { RenderType.entityCutoutNoCull(DefaultPlayerSkin.getDefaultSkin()) };// other way?
+        AtomicReference<RenderType> rendertype = new AtomicReference<>(RenderType.entityCutoutNoCull(DefaultPlayerSkin.getDefaultSkin()));// other way?
         grave.getCapability(EIOCapabilities.OWNER).ifPresent((cap) -> {
             if (cap.getUUID() != null) {
-                rendertype[0] = getRenderType(pBlockEntity.getLevel().getPlayerByUUID(cap.getUUID()));
+                // TODO: Deal with the ugly null checks...
+                Player player = Objects.requireNonNull(pBlockEntity.getLevel()).getPlayerByUUID(cap.getUUID());
```

Comment:
to my knowledge getLevel() always returns a level, when it's used ingame. The new Java NullPointerException returns more information than Objects.requireNonNull, which decreases log readability. In my opinion Objects.requireNonNull should only be called, if we call a method, which could be null, to a method, which doesn't want a null and doesn't immediatly crash, but stores the null somewhere, where it might cause a NullPointer later.

On another note: player is nullable here and if the client never has seen the player (and probably if the player isn't online anymore, but I'm not sure about that), getPlayerByUUID would return null, which would crash in the next line. I would suggest to not do a nullCheck, but to change the parameter of getRenderType to Player, let that method figure the player stuff out, if it can't find a player, return the alex/steve skin by uuid. As a permanent solution for the skin thing for offline players, the whole GameProfile needs to be serialized in the Owner-Capability similar to the skull block